extensions/objects/templates/filepath-editor.component.spec.ts

```diff
@@ -0,0 +1,2312 @@
+      'pNotAk2nunvXeba05770hzI7MXbrPPae+d9t5trjntvSPNae8daU5772hzQqHQJA' +
+      'pNotAkCk2i0CQKTaLQJApNotAkCk2i0CQKTaLQJApNotAkCk2i0CQKTaLQJApNot' +
+      'AkCk2ivXekOZ3RJApNotAkCk2i0CQKTaLQJApNotAkCk2i0CQKTaLQJApNotAkCk' +
+      '2i0CQKTaLQJApNotAkCk2i0CTae6e9d5trTnvvSHMjsxdus89p75323m2uOe29I8' +
+      '1p7x1pTnvvaHNCodAkCk2i0CQKTaLQJApNotAkCk2i0CQKTaLQJApNotAkCk2i0C' +
+      'QKTaLQJApNotAkCk2i0CQKTaK9d6Q5ndEkCk2i0CQKTaLQJApNotAkCk2i0CQKTa' +
+      'LQJApNotAkCk2i0CQKTaLQJApNotAkCk2i0CQKTaLQJNp7p713m2tOe+9IcyOzF2' +
+      '6zz2nvnfbeba457b0jzWnvHWlOe+9oc0Kh0CQKTaLQJApNotAkCk2i0CQKTaLQJA' +
+      'pNotAkCk2i0CQKTaLQJApNotAkCk2i0CQKTaLQJApNMrJ770q/NAlzwaI8WWvaIb' +
+      'elWB3rodmdPllbiRtW5vUsSpYbq9uPbouGd19Ywj7OS50b2TO66yb7azAf5YB1cG' +
+      'VruVNPEYp8/nhrNlnJrZ0X3e5EMb1sETIEH+ARJfnlPu71xK/9Ll4f7bVifHCErm' +
+      'gBL22P63ZsLFlsbCtWj7STqG/FdDhd+pONKRybhEfpXM+nI/t8RvaMNjcOlIYyB4' +
+      'Bi96lG8tJn9rUmG8eO29ZtSWS+eXChndkVZRezNWAlCaGenvlNrZEzsmeAa2D+mo' +
+      'WVN0fzVx6uz5UvL5grv3ZjPZ46YklzamXl9NzO1bj5P53ZK049N3v9iXB67JXudZ' +
+      'f7oL+PkQ3daeWf2DPfuPmP33M0u+1zz/1Jpzz4M9/MX9r+C8/ur9/9i//wVLf5kf' +
+      'vWHvjL3jumFr5+wz1Pf2einGi+7aHotT/w62RkLx0t9J5vfPmvj//ZKx//SvKxR+' +
+      '95/MHmj+/8+ck7/mnHZ377vx74xn2P3/rU/L/83e/vffmZm558xxPf/oNHZn/rp7' +
+      '/eR3a5j3s9I3tGm2vPve0OPP8EfnLX9L75D149+8nx9lx8y1df2nd8/qqD+xq9Q8' +
+      'cXk/eHRz+06+3px4q7l178ue8+mjd0jd4ot9ouv7eaHJg52MN4ue1bW/fVQ7EYN2' +
+      'euf+iFT08lD2957lutrStHdn7xWBf7d+zvtxFWrHpr1Cga2dBrbRz77kL+4id/92' +
+      'Re+4n//tST394/NvXk4e7PPnVf9+r3PPapyUM/9h/jN91/7b/u+psvtm9cO5ZuP4' +
+      'k3Z7VyVEuPbOgmwtytP+r+8Mbsrbhzyz/O7z5au+uGve+97U3XvpDe/var9iy/+6' +
+      '5nJx/+i6VHbn/+hdtuefZme+DaiQ9MjO73HLe4zicu94/gwOIn7ryjO4blcQB/9F' +
+      'ijdcuDcydmugk69e5U33fKK7+zdfYkXHN5BujVkaU4Nns5jvPS5kb2jAamPnRFPA' +
+      'aMoVic+c13feO621fStLHSuaI+6RbG4tkTR67v7N2VudMxOs16Hmrdxqzl0ai+wB' +
+      'vZLzUsYdcYUGANcQOtuXt3h8OZW4qa4fQJ1/SnXmnUOsVM8/TadD+0HJIceQH4Ue' +
+      '08wqEnukBAjlYXcR/eF92tzbTvvZtwiF17vESnbKzMJD72wFLeRjsCEF7z475ORj' +
+      'b0EjxQWGMVPTRiuF6zjE/5lgsrwdfcWij7tTTYSlY72kCnnCjQ9S64eGSfz8heoy' +
+      'dQA1KgHbYgRAC8wVkG74E1pHBxhrhsWD6Wox6hgcYInzUY7WPbVBSaRKFJFJpEoU' +
+      'kUmkShSf4PLtcFn9vdh74AAAAldEVYdGRhdGU6Y3JlYXRlADIwMjEtMDctMDJUMT' +
+      'A6MjE6MzkrMDA6MDDnnN2jAAAAJXRFWHRkYXRlOm1vZGlmeQAyMDIxLTA3LTAyVD' +
+      'EwOjIxOjM5KzAwOjAwlsFlHwAAAABJRU5ErkJggg==',
+    originalWidth: 360,
+    originalHeight: 360,
+    savedImageFilename: 'saved_file_name.gif',
+    savedImageUrl: 'assets/images'
+  };
+
+  let localConvertImageDataToImageFile = (dataURI) => {
+    var byteString = atob(dataURI.split(',')[1]);
+    var mimeString = dataURI.split(',')[0].split(':')[1].split(';')[0];
+    var ab = new ArrayBuffer(byteString.length);
+    var ia = new Uint8Array(ab);
+    for (var i = 0; i < byteString.length; i++) {
+      ia[i] = byteString.charCodeAt(i);
+    }
+    var blob = new Blob([ab], {type: mimeString});
+    return blob;
+  };
```

Comment:
Why do we require this logic in the tests? Can you please leave a comment?

Reply:
Done